### PR TITLE
Galaxy bugfix

### PR DESCRIPTION
**Pull request by BlueBasilisk**
This is a small bugfix;
I thought the galaxy tools got a string in, but it actually already gets converted to a Path in immuneMLApp.py. Therefore, Galaxy currently fails due to an assertion error. This has been corrected here (tested successfully on local galaxy instance)

### CODE REVIEW

File: source/api/galaxy/Util.py [hunk 1]
```diff
@@ -11,12 +12,11 @@
 class Util:
 
     @staticmethod
-    def check_parameters(yaml_path: str, output_dir: str, kwargs, location):
+    def check_parameters(yaml_path: Path, output_dir: Path, kwargs, location):
```
Comment (pavlovicmilena) [inline source/api/galaxy/Util.py:15]: is this supposed to be like this? asserting that it's the file with os.path?

Comment (BlueBasilisk) [reply source/api/galaxy/Util.py:15]: os.path works with path objects as well as string, but I can add an assert that it should be a Path object

**Comment by BlueBasilisk** [comment]
Additional bugfixes added later:
- when accessing RepertoireDataset.metadata_file, this went wrong in galaxy because all files are replaced to a new folder. Due to this, it was impossible to run for example MatchedRegex encoder via Galaxy (which I was adding as an example to the docs). Added a check now that if the original metadata file path does not exist, but metadata file exists in the current working dir, that this then instead is returned. 
- Some more flexibility in ExploratoryAnalysisParser that if a non-ML encoding is used, Labels do not have to be defined (it is very cumbersome to have to add fake labels to the data just to run a report that doesnt use labels), and moving a defaults file for matched_regex to the correct folder

(sorry for the many e-mails, forgot to temporarily close the pull request)

**Comment by BlueBasilisk** [comment]
updated tests

File: source/data_model/encoded_data/EncodedData.py [hunk 2]
```diff
@@ -34,7 +34,8 @@ def __init__(self, examples, labels: dict = None, example_ids: list = None, feat
                     .format(len(label), len(example_ids))
                 assert examples is None or len(example_ids) == examples.shape[0], "EncodedData: there are {} example ids, but {} examples."\
                     .format(len(example_ids), examples.shape[0])
-        assert len(labels.keys()) > 0 if labels is not None else True
+        if encoding not in ["MatchedReceptorsEncoder", "MatchedSequencesEncoder", "MatchedRegexEncoder"]:
```
Comment (pavlovicmilena) [inline source/data_model/encoded_data/EncodedData.py:37]: what is happening here? :) what is different with these encoders?

File: source/data_model/dataset/RepertoireDataset.py [hunk 3]
```diff
@@ -46,14 +47,26 @@ def get_repertoire(self, index: int = -1, repertoire_identifier: str = "") -> Re
     def get_example_count(self):
         return len(self.repertoires)
 
+    def get_metadata_file(self):
```
Comment (pavlovicmilena) [inline source/data_model/dataset/RepertoireDataset.py:50]: get_metadata_file() should be a fix for galaxy issues when metadata is in the same folder? If that’s the case, would it not be better to fix that when importing data vs fixing it every time in the RepertoireDataset class? :)

**Review by pavlovicmilena — COMMENTED** [review]
Looks good, had just to minor questions :)

File: source/data_model/encoded_data/EncodedData.py (hunk 2 above)
Comment (BlueBasilisk) [reply source/data_model/encoded_data/EncodedData.py:37]: These encoders are not used for machine learning. It is annoying to have to add fake labels to a dataset without labels just to run an encoding & report on it that do not use those labels

**Comment by BlueBasilisk** [comment]
Moved the check for metadata_file path to PickleImport. The metadata file is only checked to be in the current working dir if metadata_file is not explicitly set as a dataset import param.

File: source/data_model/dataset/RepertoireDataset.py (hunk 3 above)
Comment (BlueBasilisk) [reply source/data_model/dataset/RepertoireDataset.py]: updated now

File: source/data_model/encoded_data/EncodedData.py (hunk 2 above)
Comment (BlueBasilisk) [reply source/data_model/encoded_data/EncodedData.py]: Actually thinking about it again, if someone uses ExploratoryAnalysis labels should be entirely optional. Someone could also want to get out data encoded in a different way without having labels in the dataset

**Review by pavlovicmilena — APPROVED** [review]
Looks good, just remove all the debugging print statements in PickleImport and MatchedRegexRepertoireEncoder 😁

**Comment by BlueBasilisk** [comment]
oops! debug prints are removed now :D

**Review by pavlovicmilena — APPROVED** [review]

**Pull request merged**